Loop/Managers/SupportManager.swift

```diff
@@ -0,0 +1,258 @@
+//
+//  SupportManager.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 9/8/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import Combine
+import Foundation
+import LoopKit
+import LoopKitUI
+import SwiftUI
+import MockKitUI
+
+public final class SupportManager {
+    
+    private lazy var log = DiagnosticLog(category: "SupportManager")
+
+    private var supports = Locked<[String: SupportUI]>([:])
+    
+    private var identifierWithHighestVersionUpdate: String? {
+        get {
+            return UserDefaults.appGroup?.identifierWithHighestVersionUpdate
+        }
+        set {
+            UserDefaults.appGroup?.identifierWithHighestVersionUpdate = newValue
+        }
+    }
+    
+    private let alertIssuer: AlertIssuer
+    private let pluginManager: PluginManager?
+    private let staticSupportTypes: [SupportUI.Type]
+    private let staticSupportTypesByIdentifier: [String: SupportUI.Type]
+
+    lazy private var cancellables = Set<AnyCancellable>()
+
+    init(pluginManager: PluginManager? = nil,
+         deviceDataManager: DeviceDataManager? = nil,
+         servicesManager: ServicesManager? = nil,
+         staticSupportTypes: [SupportUI.Type]? = nil,
+         alertIssuer: AlertIssuer) {
+        
+        self.alertIssuer = alertIssuer
+        self.pluginManager = pluginManager
+        self.staticSupportTypes = staticSupportTypes ??
+            (FeatureFlags.allowSimulators ? [MockSupport.self] : [])
+        staticSupportTypesByIdentifier = self.staticSupportTypes.reduce(into: [:]) { (map, type) in
+            map[type.supportIdentifier] = type
+        }
+
+        restoreState()
+
+        ((pluginManager?.availableSupports ?? [SupportUI]()) +
+         (deviceDataManager?.availableSupports ?? [SupportUI]()) +
+         (servicesManager?.availableSupports ?? [SupportUI]()) +
+         self.staticSupportTypes.map { $0.init(rawState: [:]) }.compactMap { $0 })
+            .forEach {
+                addSupport($0)
+            }
+                
+        // Perform a check every foreground entry and every loop
+        NotificationCenter.default.publisher(for: UIApplication.willEnterForegroundNotification)
+            .sink { [weak self] _ in
+                self?.performCheck()
+            }
+            .store(in: &cancellables)
+        
+        NotificationCenter.default.publisher(for: .LoopCompleted)
+            .sink { [weak self] _ in
+                self?.performCheck()
+            }
+            .store(in: &cancellables)
+    }
+}
+
+// MARK: Manage list of Supports
+extension SupportManager {
+    func addSupport(_ support: SupportUI) {
+        supports.mutate {
+            if $0[support.identifier] == nil {
+                $0[support.identifier] = support
+                support.setAlertIssuer(alertIssuer: alertIssuer)
```

Comment:
As I indicate in the other PR, I think it would be better to use a delegate (like device managers). Have the delegate be the SupportManager.